### PR TITLE
ENT-9229: Added a new API endpoint to get admin analytics aggregated data on user enrollment and engagement.

### PR DESCRIPTION
**Pull request by saleem-latif**
__Jira Ticket:__ [ENT-9229](https://2u-internal.atlassian.net/browse/ENT-9229)

__Description:__
This endpoint will return aggregated statistics for enterprise learners

1. Enrollment count in the given date range
2. Distinct course count for the enrollments in the given date range
3. Daily session average
4. Learning hours
5. Completions count
6. Last updated at (Date time when the data was last updated)
7. min enrollment date 
8. max enrollment date

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-analytics-data-api doesn't install it
    - `test-master.in` if edx-analytics-data-api pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the edx-analytics-data-api requirements installed or if you checked out edx-enterprise-data into the src directory used by edx-analytics-data-api, you can run this command through an edx-analytics-data-api shell.
        - It would be `./manage.py makemigrations` in the shell.
- [ ] [Version](https://github.com/openedx/edx-enterprise-data/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/openedx/edx-enterprise-data/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise-data/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise-data), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise-data/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-analytics-data-api](https://github.com/openedx/edx-analytics-data-api) to upgrade dependencies (including edx-enterprise-data)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-analytics-data-api will look for the latest version in PyPi.
    - Note: the edx-enterprise-data constraint in edx-analytics-data-api **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise_data/admin_analytics/data_loaders.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    engagement['activity_date'] = engagement['activity_date'].astype('datetime64[ns]')
+
+    return engagement
+
+
+def fetch_max_enrollment_datetime():
+    """
+    Fetch the latest created date from the enterprise_learner_enrollment table.
+    """
+    query = "SELECT MAX(created) FROM enterprise_learner_enrollment"
```
Comment (irfanuddinahmad) [inline enterprise_data/admin_analytics/data_loaders.py:133]: I think all rows have the same date ...

**Review by irfanuddinahmad — APPROVED** [review]
LGTM! ... once green

**Comment by saleem-latif** [comment]
@muhammad-ammar I am facing this issue where latest version of numpy and pandas are not compatoble for python `3.8` and numpy and pandas versions compatible with `3.8` and not supported on `3.12`.
Can we remove `3.8` from the supported python versions? or is there a reason we are supporting an older version?

**Comment by saleem-latif** [comment]
I see the `analyticsapi` is on 3.8, Maybe we can drop support for 3.12?

@UsamaSadiq Can you take a look at this?

**Comment by UsamaSadiq** [comment]
I believe it would be better to pick up https://github.com/edx/edx-analytics-data-api/pull/1 and complete the Python 3.12 upgrade instead of dropping the support.

**Comment by saleem-latif** [comment]
I have created tickets https://2u-internal.atlassian.net/browse/ENT-9285 and https://2u-internal.atlassian.net/browse/ENT-9286 for this work. Hopefully, the team will have the capacity to pick these up in the coming sprint.

**Pull request merged**